graphql_ppx.opam

```diff
@@ -0,0 +1,24 @@
+opam-version: "2.0"
+maintainer: "Tomasz Cichocinski <tomaszcichocinski@gmail.com>"
+authors: "Tomasz Cichocinski <tomaszcichocinski@gmail.com>"
+homepage: "https://github.com/reasonml-community/graphql_ppx"
+bug-reports: "https://github.com/reasonml-community/graphql_ppx/issues"
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+]
+
+depends: [
+  "alcotest" {with-test}
+  "cppo" {build}
+  "dune" {build}
+  "ocaml" {>= "4.06.0"}
+  "ocaml-migrate-parsetree" {build & >= "1.4.0"}
+  "ppx_tools_versioned" {build & >= "5.2.3"}
+  "reason" {build}
+  "result" {build}
+  "yojson" {build}
+]
```

Comment:
thanks! I think we wanna remove all of these `build` constraints.

LGTM otherwise.

Reply:
Hmm, the `dune` files don't declare any `ppx_runtime_libraries`, so that's why I figured it was only build-time dependencies. I'll take a look and update.

Reply:
At least dune needs not to be a build dependency, given https://github.com/ocaml/opam/issues/3850.

Otherwise maybe you're correct. I don't fully understand how build dependencies work in PPX. You need some dependencies to build the PPX but it is then used as build dependency in other projects.

Reply:
For simplicity, I've just removed all `build` constraints.